libavutil/hwcontext_qsv.c

```diff
@@ -1052,22 +1108,31 @@ static int qsv_transfer_data_to(AVHWFramesContext *ctx, AVFrame *dst,
 
     if (src->height & 15 || src->linesize[0] & 15) {
         realigned = 1;
-        memset(&tmp_frame, 0, sizeof(tmp_frame));
-        tmp_frame.format         = src->format;
-        tmp_frame.width          = FFALIGN(src->width, 16);
-        tmp_frame.height         = FFALIGN(src->height, 16);
-        ret = av_frame_get_buffer(&tmp_frame, 0);
-        if (ret < 0)
+        if (tmp_frame->format != src->format ||
+            tmp_frame->width  != FFALIGN(src->width, 16) ||
+            tmp_frame->height != FFALIGN(src->height, 16)) {
+            av_frame_unref(tmp_frame);
+
+            tmp_frame->format = src->format;
+            tmp_frame->width  = FFALIGN(src->width, 16);
+            tmp_frame->height = FFALIGN(src->height, 16);
```

Comment:
If CU is 32x32 (or 64x64), is it enough to align width and height to a multiple of 16 ?

Reply:
According to [https://github.com/Intel-Media-SDK/MediaSDK/blob/master/doc/mediasdk-man.md#mfxframeinfo](https://github.com/Intel-Media-SDK/MediaSDK/blob/master/doc/mediasdk-man.md#mfxframeinfo ) 
"Width and height of the video frame in pixels; Width must be a multiple of 16. Height must be a multiple of 16 for progressive frame sequence and a multiple of 32 otherwise."

Reply:
I should change height to be aligned to 32 in case the frame is not progressive.

Reply:
I guess this is the requirement for h264/mpeg2.

Reply:
https://github.com/Intel-Media-SDK/MediaSDK/blob/53c3676bd627a169228caaa2e15e270f60e88f13/samples/sample_encode/src/pipeline_encode.cpp#L561
The sample_encode is also wrote following this requirement.

Reply:
Is it a bug for sample_encode?

Reply:
According to information from Dima. I check the PicWidthInLumaSamples and PicHeightInLumaSamples. If they are not set  they will equal to mfxFrameInfo->width/height, so align to 16 is enough.